### PR TITLE
OPERATOR-517: fetch manifest from proxy

### PR DESCRIPTION
**Pull request by haibinxie**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: drivers/storage/portworx/manifest/common.go [hunk 1]
```diff
@@ -3,20 +3,54 @@ package manifest
...
+	if proxy == "" {
+		resp, err = httpGet(manifestURL)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Add the http prefix so we can parse the URL:
+		// Wrong format: url.Parse("127.0.0.1:3213")
+		// Correct format: url.Parse("http://127.0.0.1:3213")
+		if !strings.HasPrefix(strings.ToLower(proxy), httpPrefix) {
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/manifest/common.go:33]: Shouldn't we look for https proxy as well?

File: drivers/storage/portworx/manifest/remote_test.go [hunk 2]
```diff
@@ -8,11 +8,41 @@ import (
...
 
 	version "github.com/hashicorp/go-version"
+	"github.com/libopenstorage/operator/drivers/storage/portworx/util"
 	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
 	"github.com/stretchr/testify/require"
 	"k8s.io/api/core/v1"
 )
 
+func TestRemoteManifestWithProxy(t *testing.T) {
+	// Do not run the test by default. Change httpProxy below to a working proxy.
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/manifest/remote_test.go:18]: For unit tests, we want it to run on any environment. Let's test the proxy only in functional tests. 
For this unit test, see if you can test the behavior. May be check if http client is called with the proxy. 
If it is getting too complicated, we can skip the unit test.

**Comment by piyush-nimbalkar** [comment]
@haibinxie, sorry for the delay in review.

File: drivers/storage/portworx/manifest/remote.go [hunk 3]
```diff
@@ -65,7 +65,7 @@ func (m *remote) downloadVersionManifest(
 	params.Add("kbver", m.k8sVersion.String())
 	u.RawQuery = params.Encode()
 
-	body, err := getManifestFromURL(u.String())
+	body, err := getManifestFromURL(u.String(), pxutil.GetPxProxyEnvVarValue(m.cluster))
```
Comment (pure-jliao) [inline drivers/storage/portworx/manifest/remote.go:68]: this function can return https first if exist, we should handle https when parsing the url as well

**Review by pure-jliao — COMMENTED** [review]
one comments regarding https proxy

File: drivers/storage/portworx/manifest/remote_test.go (hunk 2 above)
Comment (haibinxie) [reply drivers/storage/portworx/manifest/remote_test.go:18]: this unit test works and talks to real proxy, but I just dont enable it by default as the proxy is setup manually.

Comment (piyush-nimbalkar) [reply drivers/storage/portworx/manifest/remote_test.go:18]: I see, so you are keeping it to test this manually only. We won't be running it any environment, correct?

Comment (haibinxie) [reply drivers/storage/portworx/manifest/remote_test.go:18]: Yes

File: drivers/storage/portworx/manifest/remote.go (hunk 3 above)
Comment (haibinxie) [reply drivers/storage/portworx/manifest/remote.go:68]: this string is later used by url.Parse which works for https. if that's where you meant to handle

File: drivers/storage/portworx/manifest/common.go (hunk 1 above)
Comment (haibinxie) [reply drivers/storage/portworx/manifest/common.go]: good point, resolved in a new commit. pls take another look

**Review by pure-jliao — APPROVED** [review]
LGTM

**Review by piyush-nimbalkar — APPROVED** [review]

**Pull request merged**